mslib/utils/airdata.py

```diff
@@ -173,41 +192,55 @@ def get_airspaces(countries=[]):
     _airspaces_mtime = {}
     _airspaces = []
     for file in files:
-        fpath = os.path.join(MSUI_CONFIG_PATH, file)
-        airspace = etree.parse(fpath)
-        airspace = airspace.find("AIRSPACES")
-        if airspace is not None:
-            for elem in airspace:
-                try:
-                    airspace_data = {
-                        "name": elem.find("NAME").text,
-                        "polygon": elem.find("GEOMETRY").find("POLYGON").text,
-                        "top": float(elem.find("ALTLIMIT_TOP").find("ALT").text),
-                        "top_unit": elem.find("ALTLIMIT_TOP").find("ALT").get("UNIT"),
-                        "bottom": float(elem.find("ALTLIMIT_BOTTOM").find("ALT").text),
-                        "bottom_unit": elem.find("ALTLIMIT_BOTTOM").find("ALT").get("UNIT"),
-                        "country": elem.find("COUNTRY").text
-                    }
-                except TypeError as ex:
-                    logging.debug("Problem %s in airspaces file %s", (ex, fpath))
-                    logging.info("A few data of %s is ignored because of an incompatible format.", fpath)
-                    continue
-
-                # Convert to kilometers
-                airspace_data["top"] /= 3281 if airspace_data["top_unit"] == "F" else 32.81
-                airspace_data["bottom"] /= 3281 if airspace_data["bottom_unit"] == "F" else 32.81
-                airspace_data["top"] = round(airspace_data["top"], 2)
-                airspace_data["bottom"] = round(airspace_data["bottom"], 2)
-                airspace_data.pop("top_unit")
-                airspace_data.pop("bottom_unit")
-
-                airspace_data["polygon"] = [(float(data.split(" ")[0]), float(data.split(" ")[-1]))
-                                            for data in airspace_data["polygon"].split(",")]
-                _airspaces.append(airspace_data)
-                _airspaces_mtime[file] = os.path.getmtime(os.path.join(MSUI_CONFIG_PATH, file))
-        else:
-            QtWidgets.QMessageBox.information(None,
-                                              "No Airspace data in file:",
-                                              f"{file}")
+        fpath = os.path.join(osdir, file)
```

Comment:
This looks very different. I do not fully understand the minidom, but I somehow lack the reference to the AIRSPACES tag. Why is this not necessary anymore?

Reply:
I found after I used minidom a typo in my example.  May be I could also improve it now for ET.  I check this.

The info in that message box is misleading, AIRSPACES Tag is always there, an empty dataset looks like

```
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<OPENAIP VERSION="1669769794284" DATAFORMAT="1.1" xmlns="https://www.openaip.net" loremm ipsum ...>
 <AIRSPACES></AIRSPACES>
</OPENAIP>
~         
```
I will add an explicit check I do it currently by enumerate(names) which is empty, when there is no data.